utils/chunked_vector.hh

```diff
@@ -142,10 +138,16 @@ public:
     }
     template <typename... Args>
     T& emplace_back(Args&&... args) {
-        reserve_for_push_back();
-        auto& ret = *new (addr(_size)) T(std::forward<Args>(args)...);
+        pointer ret;
+        if (_capacity > _size) {
+            ret = new (addr(_size)) T(std::forward<Args>(args)...);
+        } else {
+            auto x = get_chunk_before_emplace_back();
```

Comment:
1. the chunk returned by `get_chunk_before_emplace_back` is empty of items so we can lose it safely on exception, even if new item ctor failed.  The existing items are migrated to the new chunk only by `set_chunk_after_emplace_back`, which is marked noexcept exactly for this reason.  This is possible since we require the values to be nothrow-move-constructible
2. chunked_vector allocates smaller chunks initially.  Only after filling the first chunk we start allocating chunks in full-size.